mono/mini/mini-amd64.c

```diff
@@ -6880,9 +6870,12 @@ mono_arch_emit_prolog (MonoCompile *cfg)
 
 	cfg->arch.stack_alloc_size = alloc_size;
 
+	set_code_cursor (cfg, code);
```

Comment:
Some of the set_code_cursor calls are redundant.

The idea is, we optimize out the realloc checks in the existing code, so we don't need code_len to be accurate/up-to-date -- only before a realloc check. It isn't pretty, but it is incremental.

But some of the calls are required, so that realloc can use code_len.